defaults/main.yml

```diff
@@ -189,76 +195,106 @@ _nifi_backup_target: "{{ nifi_backup_target }}/nifi.{{ _nifi_backup_id }}"
 _temp_backup_dir: "/tmp/backup/nifi"
 _pre_restore_dir: "/tmp/pre_restore/nifi"
 
-_populate_backup_dir_command: "{% for source in nifi_backup_sources %}{% if loop.index > 1 %} && \\\n{% endif %}
+#
+# Backup Commands
+#
+
+_ensure_empty_backup_dir_command: rm -rf "{{ _temp_backup_dir }}" || mkdir -p "{{ _temp_backup_dir }}"
+
+_nifi_link_for_backup_command: "{% for source in nifi_backup_sources %}{% if loop.index > 1 %} && \\\n{% endif %}
    {% set source_parent = (source | dirname) %}
    mkdir -p \"{{ _temp_backup_dir }}{{ source_parent }}\" && \\\n
    ln -s \"{{ source }}\" \"{{ _temp_backup_dir }}{{ source }}\"
    {% endfor %}"
-_move_pre_restore_dirs_command: "{% for source in nifi_backup_sources %}{% if loop.index > 1 %} && \\\n{% endif %}
+
+_cleanup_backup_dir_command: rm -rf "{{ _temp_backup_dir }}"
+
+#
+# Restore Commands
+#
+
+_ensure_empty_prerestore_dir_command: rm -rf "{{ _pre_restore_dir }}" || mkdir -p "{{ _pre_restore_dir }}"
+
+_nifi_save_prerestore_command: "{% for source in nifi_backup_sources %}{% if loop.index > 1 %} && \\\n{% endif %}
```

Comment:
can we use `loop` here and below?

https://docs.ansible.com/ansible/latest/user_guide/playbooks_loops.html

Reply:
I don't think so?  These commands aren't getting run via ansible - they're getting run via a cron job much later.

I do think it'd be cool if ansible had some way of installing ansible scripts as cron jobs, but that's just an idea and definitely not supported by Stouts.backup